### PR TITLE
fix: Add missing fields to Payment

### LINKED ISSUE #2: Dependency missing from repositories?
My project's Gradle build cannot find the following dependency (written in Maven for clarity):

```
<dependency>
    <groupId>ca.paymentrails</groupId>
    <artifactId>paymentrails</artifactId>
    <version>1.0</version>
</dependency>

```
I have the mavenCentral() repository declared. Is this lib stored on some other unknown repo, perhaps?
Also, the Readme.MD file contains nothing but the following under the dependency code snippet:

> The library is hosted at [insert github link]

### PR DESCRIPTION
**Pull request by mortondaniel**
# Fixes #

Adds missing fields to Payment

### Checklist

- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, create a GitHub Issue in this repository.

### CODE REVIEW

File: src/main/java/com/trolley/trolley/Payment.java [hunk 1]
```diff
@@ -35,10 +35,25 @@ public class Payment
     private String methodDisplay;
     private Batch batch;
     private String checkNumber;
-    String withholdingAmount;
-    String withholdingCurrency;
-    String equivalentWithholdingAmount;
-    String equivalentWithholdingCurrency;
```
Comment (mortondaniel) [inline src/main/java/com/trolley/trolley/Payment.java:41]: Wasn't sure if these were intentionally package private.
Let me know if that was intentional / feel free to remove

File: src/main/java/com/trolley/trolley/Payment.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public class Payment
     private String fxRate;
     private String memo;
     private String externalId;
-    private Object processedAt;
+    private String processedAt;
```
Comment (mortondaniel) [inline src/main/java/com/trolley/trolley/Payment.java:29]: Noticed this one is actually a String in the API / webhooks.
Also wasn't sure if that was intentional or not (if you're planning to change the type)

**Comment by mortondaniel** [comment]
cc @Aman-Aalam

File: src/main/java/com/trolley/trolley/Payment.java [hunk 3]
```diff
@@ -292,18 +307,138 @@ public Batch getBatch() {
...
+    public void setBatch(final Batch batch) {
         this.batch = batch;
     }
 
     public String getCheckNumber() {
         return checkNumber;
     }
 
-    public void setCheckNumber(String checkNumber) {
+    public void setCheckNumber(final String checkNumber) {
```
Comment (mortondaniel) [inline src/main/java/com/trolley/trolley/Payment.java:318]: Wasn't sure what the convention was with adding `final` - noticed most had it, except these two.

**Review by Aman-Aalam — APPROVED** [review]

**Pull request merged**